features/nanostack/nanostack-interface/NanostackRfPhy.h

```diff
@@ -24,6 +24,13 @@
 class NanostackRfPhy : public NanostackPhy {
 public:
 
+    /** Return the default on-board NanostackRfPhy
+     *
+     * Returns the default on-board NanostackRfPhy - this will be target-specific, and
+     * may not be available on all targets.
+     */
+    static NanostackRfPhy &get_default_instance();
```

Comment:
curiosity why is this a reference and the rest pointers?

Reply:
Slightly different approach to "does it exist" handling.

This version aligns with what `EMAC::get_default_instance()` does, which says that it must exist, if `DEVICE_XXX` is defined. There's no "null" possibility, hence a reference. It's kind of the "HAL" end.

For all the app-level end - `NetworkInterface` and similar, there is no controlling macro, but there is the possibility to override at link or run time (including a runtime check for a module being fitted!). In that case, there are underlying weak definitions that return `NULL` if the thing isn't provided.

So:

    #if DEVICE_EMAC
    EMAC &emac = EMAC::get_default_instance(); // must work
    #else  
    #error "No default EMAC"
    #endif

    WiFiInterface *wifi = WiFiInterface::get_default_instance();
    if (!wifi) {
        error("Wifi not available");
    }

This doesn't quite align - the upper style is closer to what the "HAL" does, but the lower is more flexible. First approach was the initial, and is long-existing on feature-emac. We  ultimately had to do the second to solve various build problems for the higher levels, and it didn't seem worth going back to change it.